riscv/src/runtime.rs

```diff
@@ -194,31 +194,22 @@ impl Runtime {
 
     pub fn with_poseidon(mut self) -> Self {
         self.add_submachine(
-            "std::machines::hash::poseidon_gl::PoseidonGL",
+            "std::machines::hash::poseidon_gl_memory::PoseidonGLMemory",
             None,
-            "poseidon_gl",
-            [format!(
-                "instr poseidon_gl link ~> {};",
-                instr_link("poseidon_gl.poseidon_permutation", 12, 4)
-            )],
-            12,
+            "poseidon_gl(memory, split_gl)",
+            [r#"instr poseidon_gl X, Y
+                link ~> val1_col = regs.mload(X, STEP)
+                link ~> val2_col = regs.mload(Y, STEP + 1)
+                link ~> poseidon_gl.poseidon_permutation(val1_col, val2_col, STEP);"#],
```

Comment:
Is anything making sure that the memory pointers passed here are aligned?

Reply:
I'll add constraints

Reply:
added constraints, please audit